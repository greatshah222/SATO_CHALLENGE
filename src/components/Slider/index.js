import React from 'react';

import MuiSlider from '@material-ui/core/Slider';

const Slider = ({ ...props }) => <MuiSlider {...props} />;

export default Slider;
