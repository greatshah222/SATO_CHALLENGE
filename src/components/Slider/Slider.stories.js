import React from 'react';
import { storiesOf } from '@storybook/react';
import { muiTheme } from 'storybook-addon-material-ui';
import { withKnobs } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withBackgrounds } from '@storybook/addon-backgrounds';
import theme from '../../theme';

import SliderExample from './SliderExample';

export const bg = withBackgrounds([{ name: 'light grey', value: '#f2f2f2', default: true }]);

storiesOf('Slider', module)
  .addDecorator(bg)
  .addDecorator(muiTheme(Object.values(theme)))
  .addDecorator(withKnobs)
  .addDecorator(story => <div style={{ padding: '40px' }}>{story()}</div>)
  .addWithJSX('default', () => {
    return <SliderExample />;
  });
