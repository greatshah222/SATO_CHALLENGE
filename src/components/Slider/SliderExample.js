import React, { Component } from 'react';

import Slider from './index';

import Text from '../Text';
import Grid from '../Grid';

class SliderExample extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: 50,
    };
  }

  render() {
    return (
      <Grid container>
        <Grid item xs={12}>
          <Text>Value is {this.state.value}</Text>
        </Grid>

        <Grid item xs={12} style={{ paddingTop: '15px' }}>
          <Slider
            color="secondary"
            min={0}
            max={100}
            step={1}
            onChange={(e, value) => this.setState({ value })}
            value={this.state.value}
          />
        </Grid>
      </Grid>
    );
  }
}

export default SliderExample;
