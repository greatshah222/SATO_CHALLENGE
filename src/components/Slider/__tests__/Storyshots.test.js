import React from 'react';
import { shallow } from 'enzyme';
import Slider from '../index';

describe('Slider', () => {
  it('renders correctly', () => {
    const tree = shallow(<Slider />);
    expect(tree).toMatchSnapshot();
  });
});
