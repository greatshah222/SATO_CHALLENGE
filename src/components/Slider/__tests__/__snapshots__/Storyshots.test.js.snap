// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `<WithStyles(ForwardRef(Slider)) />`;
